data/sqlutil/converter.go

```diff
@@ -0,0 +1,216 @@
+				return s.Replacer.ReplaceFunc(v)
+			}
+
+			return v, nil
+		},
+	}
+}
+
+// ToConverters creates a slice of Converters from a slice of StringConverters
+func ToConverters(s ...StringConverter) []Converter {
+	n := make([]Converter, len(s))
+	for i, v := range s {
+		n[i] = v.ToConverter()
+	}
+
+	return n
+}
+
+// Converter is used to convert known types returned in sql.Row to a type usable in a dataframe.
+type Converter struct {
+	// Name is the name of the converter that is used to distinguish them when debugging or parsing log output
+	Name string
+
+	// InputScanType is the type that is used when (*sql.Rows).Scan(...) is called.
+	// Some drivers require certain data types to be used when scanning data from sql rows, and this type should reflect that.
+	InputScanType reflect.Type
+
+	// InputTypeName is the case-sensitive name that must match the type that this converter matches
+	InputTypeName string
+
+	// FrameConverter defines how to convert the scanned value into a value that can be put into a dataframe
+	FrameConverter FrameConverter
+}
+
+// DefaultConverterFunc assumes that the scanned value, in, is already a type that can be put into a dataframe.
+func DefaultConverterFunc(in interface{}) (interface{}, error) {
+	return in, nil
+}
+
+// NewDefaultConverter creates a Converter that assumes that the value is scannable into a String, and placed into the dataframe as a nullable string.
+func NewDefaultConverter(name string, nullable bool, t reflect.Type) Converter {
+	slice := reflect.MakeSlice(reflect.SliceOf(t), 0, 0).Interface()
+	if !data.ValidFieldType(slice) {
+		return Converter{
+			Name:          fmt.Sprintf("[%s] String converter", t),
+			InputScanType: reflect.TypeOf(""),
+			InputTypeName: name,
+			FrameConverter: FrameConverter{
+				FieldType: data.FieldTypeNullableString,
+				ConverterFunc: func(in interface{}) (interface{}, error) {
+					v := in.(*string)
+					return v, nil
+				},
+			},
+		}
+	}
+
+	v := reflect.New(t)
+
+	var fieldType data.FieldType
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+		fieldType = data.FieldTypeFor(v.Interface()).NullableType()
+	} else {
+		fieldType = data.FieldTypeFor(v.Interface())
+	}
+
+	if nullable {
+		if converter, ok := NullConverters[t.String()]; ok {
+			return converter
+		}
+	}
+
+	return Converter{
+		Name:          fmt.Sprintf("Default converter for %s", name),
+		InputScanType: t,
+		InputTypeName: name,
+		FrameConverter: FrameConverter{
+			FieldType:     fieldType,
+			ConverterFunc: DefaultConverterFunc,
+		},
+	}
+}
+
+var (
+	// NullStringConverter creates a *string using the scan type of `sql.NullString`
+	NullStringConverter = Converter{
+		Name:          "nullable string converter",
+		InputScanType: reflect.TypeOf(sql.NullString{}),
+		InputTypeName: "STRING",
+		FrameConverter: FrameConverter{
+			FieldType: data.FieldTypeNullableString,
+			ConverterFunc: func(n interface{}) (interface{}, error) {
+				v := n.(*sql.NullFloat64)
```

Comment:
Received a panic from what seems to be this code. `interface conversion: interface {} is *sql.NullString, not *sql.NullFloat64 stack [sql_engine.go:189 panic.go:971 iface.go:261 converter.go:178 frame.go:29 sql.go:45 sql_engine.go:258`

```suggestion
				v := n.(*sql.NullString)
```